sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -150,6 +151,10 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
     assert(ipv6_hdr != NULL);
     memcpy(&remote->addr, &ipv6_hdr->src, sizeof(ipv6_addr_t));
     remote->family = AF_INET6;
+    if (local) {
+        memcpy(&local->addr, &ipv6_hdr->dst, sizeof(ipv6_addr_t));
+        local->family = AF_INET6;
+    }
```

Comment:
Is this optimized out if `MODULE_SOCK_AUX_LOCAL` is not use. I don't believe the optimizer to be that smart. If it is not optimized out, I would prefer

```suggestion
    if (IS_USED(MODULE_SOCK_AUX_LOCAL) && local) {
        memcpy(&local->addr, &ipv6_hdr->dst, sizeof(ipv6_addr_t));
        local->family = AF_INET6;
    }
```

Reply:
You're right. I added an `assert()` to aid debugging, if someone starts using the `local` parameter without using `sock_aux_local`.